azure-functions/src/lib.rs

```diff
@@ -187,118 +180,265 @@ fn initialize_app(
-        worker_path,
+        worker_dir.join("rust_worker"),
     )
     .expect("Failed to copy worker executable");
+}
 
+fn create_worker_config_file(worker_dir: &Path, verbose: bool) {
+    let config = worker_dir.join("worker.config.json");
+    if config.exists() {
+        return;
+    }
+
+    if verbose {
+        println!("Creating worker config file '{}'.", config.display());
+    }
+
+    fs::write(
+        &config,
+        to_string_pretty(&json!(
+        {
+            "description":{
+                "language": "Rust",
+                "extensions": [".rs"],
+                "defaultExecutablePath": "workers/rust/rust_worker",
+                "arguments": ["run"]
+            }
+        }))
+        .unwrap(),
+    )
+    .unwrap_or_else(|_| panic!("Failed to create '{}'", config.display()));
+}
+
+fn delete_existing_function_directories(script_root: &Path, verbose: bool) {
     for entry in fs::read_dir(&script_root).expect("failed to read script root directory") {
         let path = script_root.join(entry.expect("failed to read script root entry").path());
         if !path.is_dir() || !has_rust_files(&path) {
             continue;
         }
 
-        println!(
-            "Deleting existing Rust function directory '{}'.",
-            path.display()
-        );
+        if verbose {
+            println!(
+                "Deleting existing Rust function directory '{}'.",
+                path.display()
+            );
+        }
 
         fs::remove_dir_all(&path)
             .unwrap_or_else(|_| panic!("Failed to delete function directory '{}", path.display()));
     }
+}
 
-    for (name, info) in registry.lock().unwrap().iter() {
-        let function_dir = script_root.join(name);
-        fs::create_dir(&function_dir).unwrap_or_else(|_| {
-            panic!(
-                "Failed to create function directory '{}'",
-                function_dir.display()
-            )
-        });
+fn create_function_directory(script_root: &Path, function_name: &str, verbose: bool) -> PathBuf {
+    let function_dir = script_root.join(function_name);
 
-        let source_file = get_source_file_path(
-            Path::new(
-                info.manifest_dir
-                    .as_ref()
-                    .expect("Functions should have a manifest directory.")
-                    .as_ref(),
-            ),
-            Path::new(
-                info.file
-                    .as_ref()
-                    .expect("Functions should have a file.")
-                    .as_ref(),
-            ),
-        );
+    if verbose {
+        println!("Creating function directory '{}'.", function_dir.display());
+    }
 
-        let destination_file = function_dir.join(
-            source_file
-                .file_name()
-                .expect("expected the source file to have a file name"),
-        );
+    fs::create_dir(&function_dir).unwrap_or_else(|_| {
+        panic!(
```

Comment:
Should remove this panic in favor of changing this to return `Result` like the other functions.